presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,245 @@
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.facebook.presto.spi.predicate.Marker.Bound.ABOVE;
+import static com.facebook.presto.spi.predicate.Marker.Bound.BELOW;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+public class ElasticsearchQueryBuilder
+{
+    private static final Logger log = Logger.get(ElasticsearchQueryBuilder.class);
+
+    private final int scrollTime;
+    private final int scrollSize;
+    private final Client client;
+    private final int shard;
+    private final TupleDomain<ColumnHandle> tupleDomain;
+    private final List<ElasticsearchColumnHandle> columns;
+    private final String index;
+    private final String type;
+
+    public ElasticsearchQueryBuilder(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        ElasticsearchTableDescription table = split.getTable();
+        tupleDomain = split.getTupleDomain();
+        index = table.getIndex();
+        type = table.getType();
+        shard = split.getShard();
+        try {
+            Settings settings = Settings.builder().put("cluster.name", table.getClusterName()).build();
+            client = new PreBuiltTransportClient(settings).addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(split.getSearchNode()), table.getPort()));
+        }
+        catch (IOException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, e.getMessage());
+        }
+        columns = columnHandles;
+        scrollTime = config.getScrollTime();
+        scrollSize = config.getScrollSize();
+    }
+
+    public SearchRequestBuilder buildScrollSearchRequest()
+    {
+        StringBuilder shardBuilder = new StringBuilder("_shards:");
+        shardBuilder.append(shard);
+        SearchRequestBuilder searchRequestBuilder = client.prepareSearch(index != null && !index.isEmpty() ? index : "_all").setTypes(type).setSearchType(SearchType.QUERY_AND_FETCH).setScroll(new TimeValue(scrollTime)).setQuery(buildSearchQuery()).setPreference(shardBuilder.toString()).setSize(scrollSize);
+        log.info("ElasticSearch Request: " + searchRequestBuilder.toString());
```

Comment:
log.info("Elasticsearch search request: %s", searchRequestBuilder); 
Do you think this should be debug instead of info? Can this flood the logs?